### ISSUE TITLE
data not on same device

### ISSUE THREAD

**Issue by liu-x-p** [BurnhamG/PyTorch-ScalableFHVAE#1]
Hi, thank you for open source your work. I was also trying to rewrite FHVAE to pytorch these days, but met some trouble.
When I run your code, I found when computed loss, some of the variables were not on same device, so I put them all on cpu to be an easy solution.
And I have a question, that is, maybe the model _simple FHVAE_ is the model without Hierarchical Sampling or there are something different with the original FHVAE?
Hope for your reply.

**Comment by BurnhamG** [comment]
@liu-x-p Thanks for taking a look at the code, and trying to run it! Unfortunately this code isn't complete - I've had to focus on other matters, and haven't been able to come back and keep working on this project. I would definitely like to come back and finish this up, but I can't make any guarantees as to when that may happen. Because it is incomplete, issues like variables on different devices are probably pretty common; I'll certainly keep an eye out for that kind of problem whenever I pick this back up.

As for your question, I'll check my notes and get back to you ASAP on the difference between the simple and normal versions of the FHVAE.

**Comment by liu-x-p** [comment]
@BurnhamG Thank you for reply. I noticed that there are something unfinished in the eval_model.py. And I compared the model with the tf-version, and found several differences. I'm not familiar with tensorflow so I met some trouble when rewriting the code into pytorch. I will continue my work according to your current code. Thank you!
